rootston/desktop.c

```diff
@@ -402,20 +412,22 @@ struct roots_subsurface *subsurface_create(struct roots_view *view,
 	return subsurface;
 }
 
-void view_finish(struct roots_view *view) {
-	view_damage_whole(view);
-	wl_signal_emit(&view->events.destroy, view);
+void view_destroy(struct roots_view *view) {
+	if (view == NULL) {
+		return;
+	}
 
-	wl_list_remove(&view->new_subsurface.link);
+	wl_signal_emit(&view->events.destroy, view);
 
-	struct roots_view_child *child, *tmp;
-	wl_list_for_each_safe(child, tmp, &view->children, link) {
-		child->destroy(child);
+	if (view->wlr_surface != NULL) {
+		view_unmap(view);
 	}
 
-	if (view->fullscreen_output) {
```

Comment:
Where is this handled now? I expected to see the output use the unmap signal, but (I may be blind) I'm missing that code.

Reply:
It's done in `view_unmap`.